lms/djangoapps/grades/api/views.py

```diff
@@ -226,3 +235,104 @@ def get(self, request, course_id, **kwargs):
         if isinstance(course, Response):
             return course
         return Response(GradingPolicySerializer(course.raw_grader, many=True).data)
+
+
+@view_auth_classes()
+class AdditionalGradeReport(GenericAPIView):
+    """
+    **Use Case**
+
+        Get the additional course grade report by requested url.
+
+    **Example requests**:
+
+        * Report by section
+        GET api/grades/v0/course_grade/{course_id}/report_by_section/
+
+        * Report by section, with up-to-date-grade of the block_id requested
+        GET api/grades/v0/course_grade/{course_id}/report_by_section/{block_id}/
+
+        * Report by assignment type
+        GET api/grades/v0/course_grade/{course_id}/report_by_assignment_type/
+
+        * Report by assignment type, with up-to-date-grade of the block_id requested
+        GET api/grades/v0/course_grade/{course_id}/report_by_assignment_type/{block_id}/
+
+        * Enhanced problem grade report
+        GET api/grades/v0/course_grade/{course_id}/report_enhanced_problem_grade/
+
+    **Response Values**
+
+        * status: Message status of the report request.
+
+        * url: The absolute url to json resource.
+    """
+    TYPE_REPORT_BY_URL_NAME = {
+        'grade_course_report_by_section': 'section_report',
+        'grade_course_report_by_assignment_type': 'section_report',
+        'grade_course_report_enhanced_problem_grade': 'enhanced_problem_report',
+    }
+
+    def get(self, request, course_id, **kwargs):
+        """
+        Public method to send a Celery task, and then, generate a JSON object representation
+        with status and url of the resource requested.
+        """
+        grade_report_info = {
+            'section_block_id': ''
+        }
+        path_url = resolve(request.path_info).url_name
+        grade_report_info['task_type'] = self.TYPE_REPORT_BY_URL_NAME[path_url]
+        if kwargs:
+            grade_report_info['section_block_id'] = kwargs['usage_key_string']
+        try:
+            grades_report_task = calculate_grades_report.apply_async(args=[course_id, grade_report_info])
+            url_from_reverse = reverse('grades_api:grade_course_report_generated', args=[grades_report_task.task_id])
+            host_url = getattr(settings, 'LMS_ROOT_URL', '')
+            resource_url = '{}{}'.format(host_url, url_from_reverse)
+            success_status = _("The grade report is being created.")
+            return Response({
+                "status": success_status,
+                "url": resource_url,
+            })
+        except AlreadyRunningError:
+            already_running_status = _(
+                "The grade report is currently being created."
+                " To view the status of the report, see next link. {}{}"
+                " You will be able to download the report when it is complete.").format(host_url, url_from_reverse)
+            return Response({"status": already_running_status})
```

Comment:
[As previously noted](https://github.com/edx-olive-oc/edx-platform/pull/5#discussion_r206780028) please set HTTP status codes on all of these responses to indicate the status of the request.  Numeric status codes makes AJAX requests easier to handle.  e.g.,

* `202 -- Accepted` for when the report is being generated, but is not yet available.
* `200 - OK` for when the report is ready for download.
* `500 - Server error` for when unexpected errors are reported